### PR TITLE
unit tests for SpacesService

### LINKED ISSUE #2296: Add unit tests for the space service
We use
https://github.com/fabric8-ui/fabric8-ui/blob/master/src/app/shared/spaces.service.ts
everywhere, hence we need to add unit tests for this.

### PR DESCRIPTION
**Pull request by harshal-dhumal**
test case to test current space update scenario.

fixes #2296
this commit does not cover all services provided by SpacesService.

### CODE REVIEW

File: src/app/shared/spaces.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+			tmpspace.name = "space_" + spaceId;
+			tmpspace.id = spaceId;
+			tmpspace.attributes.name = "space_" + spaceId;
+			return  Observable.of(tmpspace);
+    };
+
+    mockProfileService = jasmine.createSpy('ProfileService');
+    mockProfileService.current = Observable.of(profile);
+
+    TestBed.configureTestingModule({
```
Comment (bartoszmajsak) [inline src/app/shared/spaces.service.spec.ts]: How about using angular component to test service with it together? I assume this service is used by some component so we could test both the service and ui component behaviour. For this purpose you can have a look at `TestContext` which simplifies usage of `TestBed` https://github.com/fabric8-ui/fabric8-ui/commit/04d18c56ab9ef766637ef34253ad4b1a4efea6ca#diff-6f9e140c353109a019c0342cc6d44aa6R25

**Comment by bartoszmajsak** [comment]
[test]

File: src/app/shared/spaces.service.mock.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  SpaceNamePipe
+} from 'ngx-fabric8-wit';
+
+
+export const context1: Context = {
+  user: {
+    "attributes": {
+      "fullName": "Corinne Krych",
+      "imageURL": "https://www.gravatar.com/avatar/d4445894370ad3aeb93345b544ba1dbf.jpg",
+      "username": "ckrych@redhat.com"
```
Comment (joshuawilson) [inline src/app/shared/spaces.service.mock.ts]: Does @corinnekrych know that you are adding her into the code like this?

Comment (harshal-dhumal) [reply src/app/shared/spaces.service.mock.ts]: No. I referred mock data from other test cases.
Also I thought username and other credentials are just place holders for testing purpose.

**Pull request closed without merging**